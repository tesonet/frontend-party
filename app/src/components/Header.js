import React, { PropTypes } from 'react';
import FlatButton from './FlatButton';
import './Header.scss';

const Header = ({ onButtonClick }) => (
  <div className="header">
    <img src={require('../../public/images/logo.png')} alt="testio" />
    <FlatButton
      icon="log-out"
      onClick={onButtonClick}
    >
       Logout
     </FlatButton>
  </div>
);

Header.propTypes = {
  onButtonClick: PropTypes.func.isRequired,
};

export default Header;
