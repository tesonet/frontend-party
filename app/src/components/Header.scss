.header {
  flex: 0 0 120px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  padding: 40px 10px;
}
