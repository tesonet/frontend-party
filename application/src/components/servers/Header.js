import React, { Component } from 'react';
import { connect } from 'react-redux';
import { logout } from '../../actions/index';
import logoDark from '../../assets/logo-dark.png';
import './styles/HeaderStyle.css';

class LogoutIcon extends Component {
  render() {
    return(
      <svg className="logout-icon" width="16" height="16" viewBox="0 0 16 16">
        <path id="ico-logout" d="M3027,54l-4,4,4,4V59h6V57h-6V54Zm11,12h-9a0.945,0.945,0,0,1-1-1V62h2v2h7V52h-7v2h-2V51a0.945,0.945,0,0,1,1-1h9a0.945,0.945,0,0,1,1,1V65A0.945,0.945,0,0,1,3038,66Z" transform="translate(-3023 -50)"/>
      </svg>
    )
  }
}

class Header extends Component {
  render() {
    return(
      <div className='header'>
          <div className='top row'>
            <div className='col-xs-6'><img src={logoDark} alt={'Testio logo'} /></div>
            <div className='col-xs-6 right log-out'>
              <div className='' onClick={() => {this.props.logout()}}>
            <LogoutIcon />
            <span>Logout</span></div></div>
          </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    login: state.LoginReducer,
    servers: state.ServersReducer
  }
}

const mapDispatchToProps = (dispatch) => {
return {
  logout: () => {
    dispatch(logout());
  }
}
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
