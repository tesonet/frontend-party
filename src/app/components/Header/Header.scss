$color_logout: #2f3254;
$color_logout_hover: #99cc33;
$centrify: (
  display: flex,
  justify-content: space-between,
  align-items: center
);

@mixin center_items($display, $justify-content, $align-items) {
  display: $display;
  justify-content: $justify-content;
  align-items: $align-items;
}

.header-style {
  @include center_items($centrify...);
  height: 100px;
  padding: 0 1em;
}

.logout-btn {
  @include center_items($centrify...);
  min-width: 60px;
  font-size: 13px;
  color: $color_logout;
  cursor: pointer;

  &:hover {
    color: $color_logout_hover;
    fill: $color_logout_hover;
  }
}
