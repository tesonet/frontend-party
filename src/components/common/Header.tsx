import React from 'react';
import Box from '../styled/Box';
import styled from 'styled-components';
import { ReactComponent as LogoSVG } from '../../assets/logotype-testio_black.svg';

const HeaderStyled = styled(Box)`
  height: 113px;
  display: flex;
  align-items: center;
`;

const Logo = styled(LogoSVG)`
  display: block;
  height: 30px;
  margin-left: 15px;
`;

export default function Header() {
  return (
    <HeaderStyled>
      <Logo alt="Tesonet" />
    </HeaderStyled>
  );
}
