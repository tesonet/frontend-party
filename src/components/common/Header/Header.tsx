import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import styles from './Header.module.scss';
import logo from './assets/logo-testio-small.png';
import logOut from './assets/logout.svg';
import { withRouter } from 'react-router-dom';
import { ROUTES } from '../../../routes';

interface Props extends RouteComponentProps<any> {}
const HeaderComponent: React.FC<Props> = ({ history }) => {
  const onLogOut = () => {
    localStorage.removeItem('auth-token');
    history.push(ROUTES.login);
  };
  return (
    <div className={styles.container}>
      <img data-test="header-logo" src={logo} alt="Testio" />
      <button
        data-test="logout-button"
        className={styles.logoutButton}
        onClick={onLogOut}
      >
        <img src={logOut} alt="logout" />
        Logout
      </button>
    </div>
  );
};

export const Header = withRouter(HeaderComponent);
