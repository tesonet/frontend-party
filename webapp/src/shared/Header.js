import React from 'react';
import { Link } from 'react-router-dom';

import TestioLogo from '../assets/images/logo_833x215.png';

export default function Header() {
  return (
    <div className="col-auto">
      <header className="page-header navbar fixed-top bg-white py-4">
        <div className="navbar-brand">
          <img src={ TestioLogo } height="30" alt="Testio."/>
        </div>

        <ul className="navbar-nav ml-auto">
          <li className="nav-item">
            <Link className="nav-link" to="/logout">
              <span className="oi oi-account-logout" />
              <span className="label">Logout</span>
            </Link>
          </li>
        </ul>
      </header>
    </div>
  );
}
