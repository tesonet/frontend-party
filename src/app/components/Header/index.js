import React from 'react';
import { logoDark } from '../../../assets';
import { SvgIcon } from '../';
import './Header.scss';

const Header = props => (
  <div className="header-style">
    <div>
      <img src={logoDark} alt="Testio logo" style={{ width: 110 }} />
    </div>

    <div className="logout-btn" onClick={props.logoutFunc}>
      <SvgIcon iconType="logIcon" />
      Logout
    </div>
  </div>
);

export default Header;
