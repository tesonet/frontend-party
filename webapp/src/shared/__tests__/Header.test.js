import React from 'react';
import Header from '../Header';

it('renders without crashing', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toHaveLength(1);
});

it('matches the snapshot', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
