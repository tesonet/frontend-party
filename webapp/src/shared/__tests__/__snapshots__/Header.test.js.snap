// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="col-auto"
>
  <header
    className="page-header navbar fixed-top bg-white py-4"
  >
    <div
      className="navbar-brand"
    >
      <img
        alt="Testio."
        height="30"
        src="logo_833x215.png"
      />
    </div>
    <ul
      className="navbar-nav ml-auto"
    >
      <li
        className="nav-item"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/logout"
        >
          <span
            className="oi oi-account-logout"
          />
          <span
            className="label"
          >
            Logout
          </span>
        </Link>
      </li>
    </ul>
  </header>
</div>
`;
